p2p/host/basic/basic_host.go

```diff
@@ -329,32 +336,44 @@ func (h *BasicHost) background(p goprocess.Process) {
 	for {
 		select {
 		case <-ticker.C:
```

Comment:
The timeout here is too long given that we literally won't announce new addresses to anyone, even new peers, until the timeout elapses. For now, we should probably just shrink the timeout to 5-10 seconds at the very most. Ideally we'd make this entirely event driven with multiple event "sources" but that's kind of tricky.

Reply:
The tricky part is that we have other event sources:

* AutoNAT knows about our external address (although, I don't think we properly feed this through to the host yet).
* We also have _observed_ addresses. If you take a look inside the identify service in go-libp2p, we keep track of the addresses that our peers observe us using and, when enough peers observe the same external address, we advertise it.

Reply:
We've changed the timeout.